src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -231,7 +231,7 @@ private async ContractTask<ContractState> Deploy(ApplicationEngine engine, byte[
 
             NefFile nef = nefFile.AsSerializable<NefFile>();
             ContractManifest parsedManifest = ContractManifest.Parse(manifest);
-            Helper.Check(nef.Script, parsedManifest.Abi);
+            Helper.Check(new VM.Script(nef.Script, true), parsedManifest.Abi);
```

Comment:
But isn't the call to `Helper.Check(nef.Script, parsedManifest.Abi);` assumes the strict mode is set to `true`? See the https://github.com/neo-project/neo/blob/ec7ac2bbae2e8babf6cf198478f225797c49d32c/src/Neo/SmartContract/Helper.cs#LL68C14-L68C14

It calls `Check(new Script(script, true), abi);`.

Reply:
But https://github.com/neo-project/neo/blob/ec7ac2bbae2e8babf6cf198478f225797c49d32c/src/Neo/SmartContract/Native/ContractManagement.cs#L298 it calls https://github.com/neo-project/neo/blob/ec7ac2bbae2e8babf6cf198478f225797c49d32c/src/Neo/SmartContract/Helper.cs#L79 instead of https://github.com/neo-project/neo/blob/ec7ac2bbae2e8babf6cf198478f225797c49d32c/src/Neo/SmartContract/Helper.cs#L68

Reply:
As @Liaojinghui said, the default conversion from byte to Script is non-strict, I think that was the problem.

Reply:
We can also change this to true, and looking change the false ones to use the constructor argument.

Reply:
You're right, resolved.